### PR TITLE
Update generated headers and clang format.

### PR DESCRIPTION
**Pull request by qihqi**
title:

Test plan: CI

### CODE REVIEW

File: test/cpp/jit/test_save_load.cpp [hunk 1]
```diff
@@ -13,6 +15,20 @@
 namespace torch {
 namespace jit {
 
+namespace {
+
+static Module roundtrip(const Module& m) {
```
Comment (iseeyuan) [inline test/cpp/jit/test_save_load.cpp:20]:
```suggestion
static Module roundtripThroughMobile(const Module& m) {
```

File: test/cpp/jit/test_save_load.cpp [hunk 2]
```diff
@@ -13,6 +15,20 @@
...
+  ExtraFilesMap files;
+  std::vector<IValue> constants;
+  jitModuleToPythonCodeAndConstants(m, &files, &constants);
+  CompilationOptions options;
+  mobile::Module mobilem = jitModuleToMobile(m, options);
```
Comment (iseeyuan) [inline test/cpp/jit/test_save_load.cpp:25]: Could we also save the extra files to mobile module to have a more complete roundtrip?

File: torch/csrc/jit/serialization/import.h [hunk 3]
```diff
@@ -98,5 +98,11 @@ TORCH_API Module load(
     c10::optional<c10::Device> device,
     ExtraFilesMap& extra_files);
 
+TORCH_API Module jitModuleFromSourceAndConstants(
```
Comment (gmagogsfm) [inline torch/csrc/jit/serialization/import.h:101]: Why do we need to emphasize it is JIT module when this is in torch::jit?

File: torch/jit/_serialization.py [hunk 4]
```diff
@@ -182,3 +182,72 @@ def validate_map_location(map_location=None):
         validate_cuda_device(map_location)
 
     return map_location
+
+
+def jit_module_from_flatbuffer(f):
+    try:
+        import torch._C_flatbuffer as ff
+    except ImportError:
+        print("Please include //caffe2:_C_flatbuffer as dependency.")
```
Comment (gmagogsfm) [inline torch/jit/_serialization.py:191]: Is it possible for FlatBuffer dependency to be missing in OSS? If so, then we could probably have a dedicated error message for OSS

**Review by gmagogsfm — COMMENTED** [review]
Awesome, thanks for working out a solution so quickly.

File: test/cpp/jit/test_flatbuffer.cpp [hunk 5]
```diff
@@ -1135,5 +1135,91 @@ TEST(FlatbufferTest, OperatorTest2) { // NOLINT (use =delete in gtest)
   }
 }
 
+Module jitModuleFromBuffer(void* data) {
+  auto* flatbuffer_module = mobile::serialization::GetMutableModule(data);
+  FlatbufferLoader loader;
+  mobile::Module mobilem = loader.parseModule(flatbuffer_module);
+  ExtraFilesMap files;
+  std::vector<IValue> constants;
+  loader.extractJitSourceAndConstants(&files, &constants);
```
[the remark this answers is not in the document]
Comment (qihqi) [reply test/cpp/jit/test_flatbuffer.cpp]: yes.

File: test/cpp/jit/test_save_load.cpp (hunk 2 above)
Comment (qihqi) [reply test/cpp/jit/test_save_load.cpp]: extra files is orthogonal. 
The full roundtrip is
jit::Module -> (ivalue, source, constant) -> flatbuffer -> (ivalue, source, constant) -> jit::Module.
This test only tests first first and last step:
jit::Module -> (ivalue, source, constant)->jit::Module.

A test with extrafiles would be better aimed at testing torch::jit::load when it handles flatbuffer.

**Comment by qihqi** [comment]
@gmagogsfm  @iseeyuan as suggested I split the diff to make the generated header / clangformat change first. I'll publish another PR with the JIT source stuff.

File: tools/linter/clang_format_all.py [hunk 6]
```diff
@@ -39,6 +39,8 @@ def get_allowlisted_files() -> Set[str]:
     for dir in CLANG_FORMAT_ALLOWLIST:
         for root, dirnames, filenames in os.walk(dir):
             for filename in filenames:
+                if filename.endswith('mobile_bytecode_generated.h'):
```
[the remark this answers is not in the document]
Comment (qihqi) [reply tools/linter/clang_format_all.py]: done.

**Review by iseeyuan — APPROVED** [review]
LGTM. Please make sure the size with JIT module would be on par with the pickle format, where constants are shared and the overhead of saving JIT module is small.

**Pull request closed without merging**